### PR TITLE
search: added terms safelisting

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/inveniosoftware/invenio-records-resources/issues/447

### CODE REVIEW

**Comment by alejandromumo** [comment]
Tests might fail because it depends on https://github.com/inveniosoftware/invenio-records-resources/pull/462

File: invenio_communities/members/services/config.py [hunk 1]
```diff
@@ -55,6 +55,7 @@ class PublicSearchOptions(SearchOptions):
             "user.profile.affiliations",
         ],
         tree_transformer_factory=SearchFieldTransformer.factory(
+            allow_list=["user.profile.affiliations", "user.profile.full_name", "user.username"],
```
Comment (ppanero) [inline invenio_communities/members/services/config.py:58]: missing `user.username`?

**Review by ppanero — COMMENTED** [review]
is it worth having some tests?

**Review by ppanero — APPROVED** [review]
Discussed with Karolina IRL, @alejandromumo I'm approving this but there's a need for tests in a diff PR.

**Comment by alejandromumo** [comment]
> Discussed with Karolina IRL, @alejandromumo I'm approving this but there's a need for tests in a diff PR.

Good point, I only updated tests for the query parser. And given how many things happen in this code, maybe it's a good idea to have more tests.

**Comment by ppanero** [comment]
test passing locally, requires first users-resources and records-resources
Screenshot 2023-04-17 at 14 35 40

**Pull request merged**